packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/connection/DatabaseModelBuilderState.ts

```diff
@@ -122,11 +128,9 @@ export class DatabaseModelBuilderState {
       const newEntities: EntityChange[] = [];
       for (const entity of entities) {
         let entityChangeType: EntityChangeType;
-        if (
-          this.editorStore.graphManagerState.graph.getNullableElement(
-            entity.path,
-          ) === undefined
-        ) {
+        const graph =
+          this.assistanceLightGraph ?? this.editorStore.graphManagerState.graph;
+        if (graph.getNullableElement(entity.path) === undefined) {
```

Comment:
i dont get the point of this `assistanceLightGraph` the only place its used is here. we don't use it below when loading the entities into the graph. What exactly is it you are trying to do in context of this state.

Reply:
There is a comment in my code `'This assistanceLightGraph is used to build database models if database doesn't exist in editorStore.graphManagerState.graph'`
For this flow, database is not in the graph yet so I have to use an assistanceLightGraph that contains this database otherwise you will see the error that complaining no such db

Reply:
i saw the comment still dont understand this. if the light graph is defined you use but when calling the 
`this.editorStore.graphState.loadEntityChangesToGraph` below doesn't that load entities into the main graph?
Also we should be refactoring these states to make sense in isolation of your stepper workflow. 
Therefore if we may consider having a graph input for this state if it make sense.

Reply:
there is a mistake. saveModels() is not used. will remove the changes. 
this graph is used in previewDatabaseModels